ladybug_geometry/geometry3d/sphere.py

```diff
@@ -0,0 +1,194 @@
+                           data['radius']))
+
+    @property
+    def center(self):
+        """Center of sphere."""
+        return self._center
+
+    @property
+    def radius(self):
+        """Radius of sphere."""
+        return self._radius
+
+    @property
+    def min(self):
+        """A Point3D for the minimum bounding rectangle vertex around this geometry."""
+        return Point3D(self.center[0] - self.radius, self.center[1] - self.radius,
+                       self.center[2] - self.radius)
+
+    @property
+    def max(self):
+        """A Point3D for the maximum bounding rectangle vertex around this geometry."""
+        return Point3D(self.center[0] + self.radius, self.center[1] + self.radius,
+                       self.center[2] + self.radius)
+
+    @property
+    def diameter(self):
+        """Diameter of sphere"""
+        return self.radius * 2
+
+    @property
+    def circumference(self):
+        """Circumference of sphere"""
+        return 2 * math.pi * self.radius
+
+    @property
+    def area(self):
+        """Area of sphere"""
+        return 4 * math.pi * self.radius ** 2
+
+    @property
+    def volume(self):
+        """Volume of sphere"""
+        return 4 / 3 * math.pi * self.radius ** 3
+
+    def move(self, moving_vec):
+        """Get a sphere that has been moved along a vector.
+
+        Args:
+            moving_vec: A Vector3D with the direction and distance to move the shere.
+        """
+        return Sphere(self.center.move(moving_vec), self.radius)
+
+    def scale(self, factor, origin=None):
+        """Scale a arc by a factor from an origin point.
+
+        Args:
+            factor: A number representing how much the sphere should be scaled.
+            origin: A Point2D representing the origin from which to scale.
+                If None, it will be scaled from the World origin (0, 0, 0).
+        """
+        return Sphere(self.center.scale(factor, origin), self.radius * factor)
+
+    def intersect_plane(self, plane):
+        """Get the intersection of a plane with this Sphere object
+
+        Args:
+            plane: A Plane object
+
+        Returns:
+            Arc3D representing a full circle if it exists.
+            None if no intersection exists.
+        """
+        r = self.radius
+        pt_c = self.center
+        pt_o = plane.o
+        v_n = plane.n.normalize()
+
+        # Resulting circle radius. Radius² = r² - [(c-p).n]²
+        v_d = ((pt_c + pt_o) * v_n)
+        d = v_d.magnitude  # d = plane.distance_to_point(self._center)
+        if r < d:  # No intersection if negative
+            return None
+        cut_r = math.sqrt(r ** 2 - d ** 2)
+        if cut_r == 0:  # Tangent intersection ignored - results in a point
+            return None
+
+        # Resulting circle center point. Center_point = p - [(c-p).n]n
+        cut_center = pt_c + v_d * v_n
+        cut_plane = Plane(v_n, cut_center)
+        return Arc3D(cut_plane, cut_r)
```

Comment:
As I mentioned in my comment before this review, all of this raw math should go in the `intersection3d` module.

Reply:
Some issues, preventing this. See below for more details.